tests/traverse_util_test.py

```diff
@@ -145,6 +145,25 @@ def test_traversal_set(self):
     with self.assertRaises(ValueError):
       traversal.set([3, 4, 5], x)  # too many values
 
+  def test_flatten_dict(self):
+    xs = {'foo': 1, 'bar': {'a': 2, 'b': {}}}
```

Comment:
Given that flatten discards empty dicts (which seems correct to me, like `'b'` here), I think we should add a sentence to that effect in the flatten docstring.